### DISCUSSION TITLE
［调查］关于是否保留应用内置的帮助文档的调查

### DISCUSSION THREAD

**Discussion by HelloWRC** [ClassIsland/ClassIsland#270] [调查]
如题，目前我正在考虑移除应用内的帮助文档，并改为提供[在线文档](https://docs.classisland.tech)。主要有以下几点原因：

1. 目前随着应用功能的丰富和帮助文档的建设，帮助文档的图片逐渐增多，相应地，帮助文档的体积也在逐渐增大。移除应用内置的帮助文档后可以减小应用的打包大小。
2. 目前应用的帮助文档主要在在线文档上更新，由于在线文档和应用内置文档的Markdown语法支持状况有些差异，同步文档不是一件容易的事情，并且在应用内阅读帮助文档的体验也不如在线文档。

综合以上几点原因，我目前征求社区的意见，以决定应用内置帮助文档的去留。

投票从此讨论发布起开始，到 2024/8/21 18:00 (UTC+8) 结束。

**Comment by jiangyin14** [comment]
建议保留一个小按钮激活WebView的在线Doc

**Comment by LyCecilion** [comment]
> 大风泱泱，大潮滂滂。洪水图腾蛟龙，烈火涅槃凤凰。文明圣火，千古未绝者，惟 ClassIsland 无双；和天地并存，与日月同光。

我有这样的思考：对应用后续开发的导向也应当考虑到软件受众。ClassIsland 的受众大多是有技术的人员，最次也至少是懂得电脑基本操作的学生——基于此，不可能存在拿上应用后连基本的操作都不会的情况。

再其次，「课表」和「时间表」作为贯穿学生成长历程的两个概念，它们本身以及围绕它们衍生出的所有概念，诸如「换课」「轮换」之类，都应当为所有学生所熟知。ClassIsland 是解决实际问题的工具，而它所围绕概念为用户熟知也奠定了 ClassIsland 并不具有较高使用门槛的特性。换言之，任何一个人拿上 ClassIsland，都可以做到基础功能无障碍使用；较为复杂的功能，例如「集控」，查阅文档后也可以自行操作。

因此，个人观点是，**没有必要在应用内内置帮助文档**。但在线帮助文档仍然有存在的必要性。可以考虑在应用内可能产生困惑的地方多嵌入一些指向在线文档的超链接。
